hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.controller.js

```diff
@@ -49,36 +50,31 @@ define([
     };
     $scope.utils = utils;
 
-    angular.copy(entity, $scope.entity);
-    angular.copy(files, $scope.files);
-    $scope.entity.details.period_start_date = convertToDateObject($scope.entity.details.period_start_date);
-    $scope.entity.details.period_end_date = convertToDateObject($scope.entity.details.period_end_date);
+    $scope.cancel = cancel;
+    $scope.fileMoveToTrash = fileMoveToTrash;
+    $scope.filesValidate = filesValidate;
+    $scope.save = save;
 
     // Init
     (function init () {
-      angular.forEach($scope.files, function (entityFiles, entityName) {
-        $scope.filesTrash[entityName] = [];
-      });
+      initFormEntity();
+      initFilesAndUploaders();
 
       $modalInstance.opened.then(function () {
         $rootScope.$broadcast('hrjc-loader-hide');
       });
 
-      angular.forEach($scope.uploader, function (entity) {
-        angular.forEach(entity, function (field) {
-          field.onAfterAddingAll = function () {
-            $scope.filesValidate();
-          };
-        });
-      });
-
       $rootScope.$broadcast('hrjc-loader-show');
       fetchInsurancePlanTypes().then(function () {
         $rootScope.$broadcast('hrjc-loader-hide');
       });
     }());
 
-    $scope.cancel = function () {
+    /**
+     * Hides the job contract modal, but before doing so asks for confirmation
+     * before changes are lost.
+     */
+    function cancel () {
```

Comment:
The func name is 'cancel', but it 'Hides the job contract modal, but before doing so asks for confirmation before changes are lost.' Can we rename this func? promptAboutChangesLossAndCloseOnConfirm()

Reply:
I'll change the name of the function, but I'll leave it stored in the scope as "cancel"  (`$scope.cancel = cancel`) because that's how it's referenced on the view and the view is shared by 2 different controllers.

Reply:
As a bonus: added new tests for cancel